src/main/java/io/leangen/geantyref/VarMap.java

```diff
@@ -95,7 +102,18 @@ AnnotatedType map(AnnotatedType type, MappingMode mappingMode) {
             if (!map.containsKey(tv)) {
                 if (mappingMode.equals(MappingMode.ALLOW_INCOMPLETE)) {
                     AnnotatedTypeVariable variable = (AnnotatedTypeVariable) type;
-                    AnnotatedType[] bounds = map(variable.getAnnotatedBounds(), mappingMode);
+                    AnnotatedType[] bounds;
+                    if(!currentlyResolvingTypeVariable.add(tv)) {
+                        //Issue #27: We have already seen this TypeVariable during the current resolution, so we need to abort
+                        //Otherwise, we get a StackOverflowError
+                        bounds = new AnnotatedType[]{new AnnotatedTypeImpl(Object.class)};
+                    }else {
+                        try{
+                            bounds = map(variable.getAnnotatedBounds(), mappingMode);
+                        }finally{
+                            currentlyResolvingTypeVariable.remove(tv);
+                        }
```

Comment:
Not sure I get the point behind the `try` block?

Reply:
To cleanup the set, if a var is used in multiple places during parameter resolution, otherwise a second use to the var, would terminate too early.